### PR TITLE
Add missing docs for prebuilt_apple_framework rule

### PR DESCRIPTION
**Pull request by aledalgrande**
Docs were missing but I learned by searching around the internets the construct was available. Added minimal docs.

### CODE REVIEW

**Comment by ryu2** [comment]
cc: @robbertvanginkel @nguyentruongtho

**Comment by Coneko** [comment]
cc @yiding

File: docs/rule/prebuilt_apple_framework.soy [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  {call buck.page}
+    {param title: 'prebuilt_apple_framework()' /}
+    {param navid: 'rule_prebuilt_apple_framework' /}
+    {param prettify: true /}
+    {param description}
+      A prebuilt_apple_framework() rule represents an Apple bundle that has
+      already been compiled and exports a set of headers.
+    {/param}
+    {param content}
+      An <code>apple_library()</code> rule represents a set of Objective-C/C++
```
Comment (Coneko) [inline docs/rule/prebuilt_apple_framework.soy]: You probably meant `prebuilt_apple_framework` here.

File: docs/rule/prebuilt_apple_framework.soy [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    {param navid: 'rule_prebuilt_apple_framework' /}
+    {param prettify: true /}
+    {param description}
+      A prebuilt_apple_framework() rule represents an Apple bundle that has
+      already been compiled and exports a set of headers.
+    {/param}
+    {param content}
+      An <code>apple_library()</code> rule represents a set of Objective-C/C++ 
+      source files and is very similar to a 
+      <a href="{ROOT}rule/cxx_library.html"><code>prebuilt_cxx_library()</code></a>
```
Comment (Coneko) [inline docs/rule/prebuilt_apple_framework.soy]: The link should point to `prebuilt_cxx_library`.

File: docs/rule/prebuilt_apple_framework.soy [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{namespace prebuilt_apple_framework}
+
+/***/
+{template .soyweb}
+  {call buck.page}
+    {param title: 'prebuilt_apple_framework()' /}
+    {param navid: 'rule_prebuilt_apple_framework' /}
+    {param prettify: true /}
+    {param description}
+      A prebuilt_apple_framework() rule represents an Apple bundle that has
```
Comment (Coneko) [inline docs/rule/prebuilt_apple_framework.soy]: An Apple framework, not a bundle.

**Comment by aledalgrande** [comment]
Thanks for the review @Coneko.

**Comment by yiding** [comment]
I think the 'preferred_linkage' flag is required, where "shared" means the usual framework, while "static" is the not-quite-a-framework-that's-packaged-like-a-framework.

**Comment by aledalgrande** [comment]
I successfully used a "not-quite-a-framework-that's-packaged-like-a-framework" without the `preferred_linkage` flag, so don't think it's required. Is that what you meant?

**Comment by robbertvanginkel** [comment]
I thought I added and made `preffered_linkage` required in #1064, maybe you are using an older buck binary?

The main reason this wasn't documented is because this rule type has no project generation support yet, and the copying behaviour isn't correct. For dynamic frameworks these should end up in an apple_bundle, for static ones their resources should. I believe both are not implemented, so might be good to note that in the docs.

**Comment by aledalgrande** [comment]
Yeah, I have version `2016.11.11.01` (the one from Homebrew) and your PR was merged in after that release. Maybe time for a new one?

I will add a note about project generation support.

**Comment by LegNeato** [comment]
@aledalgrande filed https://github.com/facebook/buck/issues/1268

**Comment by aiked** [comment]
I updated the homebrew version to v2017.03.29.01.

**Comment by yiding** [comment]
I did add some project generation support for the case of real frameworks, where they will be copied into Frameworks/ directory of the bundle, and set various search paths and linker flags, like what you'd get with Xcode "embedded frameworks" feature.

**Comment by aledalgrande** [comment]
@yiding as soon as I can test the new release, I will add the project generation part.

**Comment by aledalgrande** [comment]
The docs should now be complete, let me know if they make sense. Also, not sure why the specs are not passing.

File: docs/rule/prebuilt_apple_framework.soy [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    {param description}
+      A prebuilt_apple_framework() rule represents an Apple framework that has
+      already been compiled and exports a set of headers.
+    {/param}
+    {param content}
+      {call buck.rule}
+        {param status: 'FROZEN' /}
+        {param overview}
+          A <code>prebuilt_apple_framework()</code> rule represents a set of
+          Objective-C/C++ source files and is very similar to a <a href="{ROOT}rule/prebuilt_cxx_library.html">
```
Comment (sdwilsh) [inline docs/rule/prebuilt_apple_framework.soy]: You should actually have used `{call buck.prebuilt_cxx_library /}` here.

**Pull request closed without merging**